src/test/extensions/updateHelper.spec.ts

```diff
@@ -31,177 +33,465 @@ const SPEC = {
+    source: {
+      name: "projects/firebasemods/sources/fake-official-source",
+    },
+  },
+};
 
-      const loggedLines = updateHelper.displayChangesNoInput(SPEC, newSpec);
+const REGISTRY_INSTANCE = {
+  name: "projects/invader-zim/instances/fake-official-instance",
+  createTime: "2019-05-19T00:20:10.416947Z",
+  updateTime: "2019-05-19T00:20:10.416947Z",
+  state: "ACTIVE",
+  config: {
+    name:
+      "projects/invader-zim/instances/image-resizer/configurations/95355951-397f-4821-a5c2-9c9788b2cc63",
+    createTime: "2019-05-19T00:20:10.416947Z",
+    sourceId: "fake-registry-source",
+    sourceName: "projects/firebasemods/sources/fake-registry-source",
+    source: {
+      name: "projects/firebasemods/sources/fake-registry-source",
+    },
+  },
+};
 
-      const expected = ["", "**Billing is no longer required for this extension.**"];
-      expect(loggedLines).to.eql(expected);
+describe("updateHelper", () => {
+  describe("updateFromLocalSource", () => {
+    let promptStub: sinon.SinonStub;
+    let createSourceStub: sinon.SinonStub;
+    let getInstanceStub: sinon.SinonStub;
+    beforeEach(() => {
+      promptStub = sinon.stub(prompt, "promptOnce");
+      createSourceStub = sinon.stub(extensionsHelper, "createSourceFromLocation");
+      getInstanceStub = sinon.stub(extensionsApi, "getInstance").resolves(INSTANCE);
     });
 
-    it("should display nothing if no relevant fields were changed", () => {
-      const newSpec = _.cloneDeep(SPEC);
-      newSpec.license = "drivers";
-
-      const loggedLines = updateHelper.displayChangesNoInput(SPEC, newSpec);
+    afterEach(() => {
+      promptStub.restore();
+      createSourceStub.restore();
+      getInstanceStub.restore();
+    });
 
-      const expected: string[] = [];
-      expect(loggedLines).to.eql(expected);
+    it("should return the correct source name for a valid local source", async () => {
+      promptStub.resolves(true);
+      createSourceStub.resolves(SOURCE);
+      const name = await updateHelper.updateFromLocalSource(
+        "test-project",
+        "test-instance",
+        ".",
+        SPEC,
+        SPEC.name
+      );
+      expect(name).to.equal(SOURCE.name);
     });
-  });
+    it("should throw an error for an invalid source", async () => {
+      promptStub.resolves(true);
+      createSourceStub.throwsException("Invalid source");
+      await expect(
+        updateHelper.updateFromLocalSource("test-project", "test-instance", ".", SPEC, SPEC.name)
+      ).to.to.be.rejectedWith(FirebaseError, "Unable to update from the source");
+    });
+    it("should not update if the update warning is not confirmed", async () => {
+      promptStub.resolves(false);
+      createSourceStub.resolves(SOURCE);
+      await expect(
+        updateHelper.updateFromLocalSource("test-project", "test-instance", ".", SPEC, SPEC.name)
+      ).to.be.rejectedWith(FirebaseError, "Update cancelled.");
+    });
+  }).timeout(2500);
```

Comment:
Why is this timeout needed? I've found that if timeouts are necessary, it's because some network call isn't being stubbed/mocked/nock'd correctly. I've recently forced all tests to have auth (for API calls) mocked out, so timeouts like this shouldn't be necessary. Is it possible to remove this timeout? (you can run this specific test by using `it.only` or passing the file name directly to `mocha`)

Reply:
Good catch. I believe it was probably for those auth API calls, since I removed the timeouts and the tests still passed.